demo-app/src/main/java/org/openmined/syft/demo/federated/domain/TrainingTask.kt

```diff
@@ -58,7 +60,22 @@ class TrainingTask(
             }
         }
         mnistJob.start(jobStatusSubscriber)
-        return statusPublisher.onBackpressureBuffer().firstOrError()
+        mnistJob.getStateFlow().collect { jobStatus ->
```

Comment:
Why the user needs to implement this? He should only be required to supply status subscriber